src/levelset/levelSetSegmentationObject.cpp

```diff
@@ -1796,7 +1797,26 @@ int LevelSetSegmentationBaseObject::evalPart(const std::array<double,3> &point)
  */
 std::array<double,3> LevelSetSegmentationBaseObject::evalNormal(const std::array<double,3> &point, bool signedLevelSet) const
 {
-    return _evalNormal(point, signedLevelSet);
+    // Project the point on the surface and evaluate the point-projection vector
+    std::array<double, 3> projectionPoint;
+    std::array<double, 3> projectionNormal;
+    evalProjection(point, signedLevelSet, &projectionPoint, &projectionNormal);
+
+    BITPIT_UNUSED(projectionPoint);
+
+    return projectionNormal;
+}
+
+/*!
+ * Evaluate the normal of the surface at the segment closest to the specified point.
+ *
+ * \param point are the coordinates of the point
+ * \param signedLevelSet controls if signed levelset function will be used
+ * \result The normal of the surface at the segment closest to the specified point.
+ */
+std::array<double,3> LevelSetSegmentationBaseObject::_evalNormal(const std::array<double,3> &point, bool signedLevelSet) const
+{
+    return evalNormal(point, signedLevelSet);
```

Comment:
It should be the other way around: the evalNormal function should call the _evalNormal function. Functions that start with a "_" are virtual functions that the derive class should be able to override.

```
void BaseClass::function()
{
    <code common for all classes>

    _function()   <---- Code specific for the derive class

    <code common for all classes> 
}
```

If evalNormal doesn't need to be overridden, just delete the _evalNormal function and do all the work in evalNormal.

Reply:
Or maybe keep only the \_evalNormal function in LevelSetSegmentationBaseObject and add there the new implementation.

Reply:
Done